app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/presenter/LearningPresenter.kt

```diff
@@ -3,73 +3,91 @@ package com.github.HumanLearning2021.HumanLearningApp.presenter
 import android.app.Activity
 import android.util.Log
 import android.widget.ImageView
-import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
-import com.github.HumanLearning2021.HumanLearningApp.model.Category
 import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
 import com.github.HumanLearning2021.HumanLearningApp.model.Dataset
-import com.github.HumanLearning2021.HumanLearningApp.model.Id
+import com.github.HumanLearning2021.HumanLearningApp.model.learning.LearningModel
 import com.github.HumanLearning2021.HumanLearningApp.view.learning.LearningMode
-import javax.inject.Inject
 
-class LearningPresenter @Inject constructor(
-    @Demo2Database
-    private val dbMgt: DatabaseManagement
+
+/**
+ * Presenter for the learning fragment
+ * @param dbMgt database manager used to retrieve data for learning
+ * @param learningMode learning mode. Influences which pictures are displayed for example.
+ * @param dataset dataset used for the learning
+ */
+class LearningPresenter(
+    private val dbMgt: DatabaseManagement,
+    private val learningMode: LearningMode,
+    private val dataset: Dataset,
 ) {
-    // may be set by the view
-    var learningMode = LearningMode.PRESENTATION
 
-    // must be set by the view
-    lateinit var dataset: Dataset
+    /**
+     * Underlying model for the learning
+     */
+    val learningModel = LearningModel(dataset)
 
-    private var previousCategory: Category? = null
+
+    /**
+     * Updates the model and the UI so that it is ready for the next sorting
+     * @param activity parent activity of the given views
+     * @param targetViews the views that display the target categories
+     * @param sourceView the view that displays the current picture to sort
+     */
+    suspend fun updateForNextSorting(
+        activity: Activity,
+        targetViews: List<ImageView>,
+        sourceView: ImageView
+    ) {
+        require(dataset.categories.size >= targetViews.size) {
+            "There must be enough categories in the dataset to display one category on each target" +
+                    " ImageView"
+        }
```

Comment:
Should we use a string ressource for this? Or because it is a require statement, it shouldn't be a string ressource ?

Reply:
It will result in a crash, so I don't think that translation will be our worst accessibility problem here xD